Sources/ClientRuntime/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -83,6 +124,7 @@ public class CRTClientEngine: HttpClientEngine {
     public func execute(request: SdkHttpRequest) async throws -> HttpResponse {
         let connectionMgr = try await serialExecutor.getOrCreateConnectionPool(endpoint: request.endpoint)
         let connection = try await connectionMgr.acquireConnection()
+
         self.logger.debug("Connection was acquired to: \(String(describing: request.endpoint.url?.absoluteString))")
         switch connection.httpVersion {
```

Comment:
The fact that the code for HTTP/1.1 & HTTP/2 is so close to one another makes me wonder if there's some common abstraction that both can adhere to, that would simplify this code

Reply:
Yeah i agree, but things could get complicated trying to find a common abstraction. Classic example of `duplication is far cheaper than the wrong abstraction`. IMO it's worth looking into a common abstraction as part of SRA adherence tracked by https://github.com/awslabs/aws-sdk-swift/issues/698 and https://github.com/awslabs/aws-sdk-swift/issues/702